src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/EnumTypeModelBinder.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding.Binders
+{
+    /// <summary>
+    /// <see cref="IModelBinder"/> implementation to bind models for types deriving from <see cref="Enum"/>.
+    /// </summary>
+    public class EnumTypeModelBinder : SimpleTypeModelBinder
+    {
+        private readonly bool _allowBindingUndefinedValueToEnumType;
+        private readonly Type _modelType;
```

Comment:
No need to store the `Type`. Passed in the `ModelBindingContext` every time. That is, the constructor should only pass the `modelType` parameter to `base(...)`.

Reply:
yes, true..but i didn't want to get the 'underlying' type in the code again and just accept what the binderprovider gives already